js/common/lib/tensor-factory.ts

```diff
@@ -86,6 +86,20 @@ export interface GpuBufferConstructorParameters<T extends Tensor.GpuBufferDataTy
   readonly gpuBuffer: Tensor.GpuBufferType;
 }
 
+export interface MLTensorConstructorParameters<T extends Tensor.MLTensorDataTypes = Tensor.MLTensorDataTypes>
+  extends CommonConstructorParameters<T>,
+    GpuResourceConstructorParameters<T> {
+  /**
+   * Specify the location of the data to be 'ml-tensor'.
+   */
+  readonly location: 'ml-tensor';
+
+  /**
+   * Specify the WebNN buffer that holds the tensor data.
```

Comment:
Find `buffer` in this PR, still exists many `buffer` should be changed to `MLTensor`.

Reply:
Done, I have removed all references to "buffer" in this PR outside of references to GPUBuffer or ArrayBuffer.